llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -3278,7 +3268,10 @@ NVPTXTargetLowering::LowerSTOREVector(SDValue Op, SelectionDAG &DAG) const {
   Ops.push_back(N->getOperand(0));
 
   // Then the split values
-  if (UpsizeElementTypes) {
+  if (ValVT.getVectorNumElements() > NumElts) {
+    // If the number of elements has changed, getVectorLoweringShape has upsized
```

Comment:
We have potentially three outcomes:
* NumElts didn't change. That's supposed to be handled by the `else` clause.
* NumElts decreased. We've upsized vector elements to 32 bits. This is what the true bllock does now.
* NumElts increased. That's an error, and it will currently result in us executing the `else` block, which would be a wrong thing to do.

I think we need an assert before the `if` that we do not expect NumElts to ever increase, or change `else` into a strict equality check and add `else llvm_unreachable()`;

Reply:
Done, added asserts